src/csharp/Microsoft.Spark.E2ETest/IpcTests/Sql/SparkSessionTests.cs

```diff
@@ -65,5 +68,90 @@ public void TestSignaturesV2_4_X()
+                data.Add(new GenericRow(new object[] { "Bob", 30 }));
+
+                var schema = new StructType(new List<StructField>()
+                {
+                    new StructField("Name", new StringType()),
+                    new StructField("Age", new IntegerType())
+                });
+                DataFrame df = _spark.CreateDataFrame(data, schema);
+                ValidateDataFrame(data.Select(a => a.Values), df, schema);
+            }
+
+            // Calling CreateDataFrame(IEnumerable<string> _) without schema
+            {
+                var data = new List<string>(new string[] { "Alice", "Bob" });
+                var schema = SchemaWithSingleColumn(new StringType());
+
+                DataFrame df = _spark.CreateDataFrame(data);                
+                ValidateDataFrame(data.Select(a => new object[] { a }), df, schema);
+            }
+
+            // Calling CreateDataFrame(IEnumerable<int> _) without schema
+            {
+                var data = new List<int>(new int[] { 1, 2 });
+                var schema = SchemaWithSingleColumn(new IntegerType());
+
+                DataFrame df = _spark.CreateDataFrame(data);
+                ValidateDataFrame(data.Select(a => new object[] { a }), df, schema);
+            }
+
+            // Calling CreateDataFrame(IEnumerable<double> _) without schema
+            {
+                var data = new List<double>(new double[] { 1.2, 2.3 });
+                var schema = SchemaWithSingleColumn(new DoubleType());
+
+                DataFrame df = _spark.CreateDataFrame(data);
+                ValidateDataFrame(data.Select(a => new object[] { a }), df, schema);
+            }
+
+            // Calling CreateDataFrame(IEnumerable<bool> _) without schema
+            {
+                var data = new List<bool>(new bool[] { true, false });
+                var schema = SchemaWithSingleColumn(new BooleanType());
+
+                DataFrame df = _spark.CreateDataFrame(data);
+                ValidateDataFrame(data.Select(a => new object[] { a }), df, schema);
+            }
+        }
+
+        /// <summary>
+        /// Validates the correctness of the dataframe given the input data.
+        /// </summary>
+        /// <param name="data">Given data for the dataframe</param>
+        /// <param name="df">Dataframe Object to validate</param>
+        /// <param name="schema">Expected schema of the dataframe</param>
+        private void ValidateDataFrame(IEnumerable<object[]> data, DataFrame df, StructType schema)
+        {
+            Assert.Equal(schema, df.Schema());
+            Row[] rows = df.Collect().ToArray();
+            Assert.Equal(data.Count(), rows.Length);
+            int i = 0;
+            foreach (object[] expectedValues in data)
+            {
+                Assert.Equal(expectedValues, rows[i].Values);
+                ++i;
+            }
+        }
+
+        /// <summary>
```

Comment:
Can you remove the comments? It's clear enough (to me it's just a distraction). We require comments in this format for public functions, but if the functions are self-describing and private, it's not necessary (unless reviewers explicitly ask for one).

Reply:
Gotcha. Cool, will make the change. Thanks!